storage/multi_iterator.go

```diff
@@ -14,13 +14,13 @@ func NewMultiIterator(iters []Iterator) Iterator {
 	return &multiIterator{current: 0, iters: iters}
 }
 
-func (m *multiIterator) Next() bool {
-	next := m.iters[m.current].Next()
-	if !next && len(m.iters)-1 > m.current {
-		m.current++
+func (m *multiIterator) Next() (next bool) {
+	for ; m.current < len(m.iters); m.current++ {
 		next = m.iters[m.current].Next()
```

Comment:
> I think you should first call Next() on the m.current partition.

That's exactly what it does do (`m.current` only gets incremented when `next` is `false`). I updated the test to have values only in storages 0 & 2, it still passes.